### PR TITLE
CHC solver: extract solver progress reporting

### PR DESCRIPTION
**Pull request by kroening**
This extracts the functionality for reporting solver progress into a separate class.

- [X] Each commit message has a non-empty body, explaining why the change was made.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [X] My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: src/cprover/solver_progress.cpp [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+void solver_progresst::finished()
+{
+  if(verbose)
+  {
+  }
+  else
+  {
+    if(consolet::is_terminal())
+    {
```
Comment (tautschnig) [inline src/cprover/solver_progress.cpp:51]: What is the reason for not doing `if(!verbose && consolet::is_terminal())` here?

File: src/cprover/solver_progress.cpp [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  else
+  {
+    if(first)
+    {
+      first = false;
+    }
+    else
+    {
+      if(consolet::is_terminal())
+        std::cout << "\x1b[1A\x1b[0K"; // up one row and clear the line
```
Comment (tautschnig) [inline src/cprover/solver_progress.cpp]: Shouldn't these also go in `consolet` as constants?

**Review by tautschnig — APPROVED** [review]

File: src/cprover/solver_progress.cpp (hunk 1 above)
Comment (kroening) [reply src/cprover/solver_progress.cpp:51]: I'll eventually get around to adding sensible output for the `--verbose` case!

File: src/cprover/solver_progress.cpp (hunk 2 above)
Comment (kroening) [reply src/cprover/solver_progress.cpp]: Yes, will do.

**Pull request merged**